apps/prairielearn/src/lib/externalImageCaptureSocket.ts

```diff
@@ -0,0 +1,79 @@
+import type { Namespace, Socket } from 'socket.io';
+
+import { logger } from '@prairielearn/logger';
+import * as Sentry from '@prairielearn/sentry';
+import { checkSignedToken } from '@prairielearn/signed-token';
+
+import { config } from './config.js';
+import type { StatusMessage } from './externalImageCaptureSocket.types.js';
+import * as socketServer from './socket-server.js';
+
+let namespace: Namespace;
+
+export function init() {
+  namespace = socketServer.io.of('/external-image-capture');
+  namespace.on('connection', connection);
+}
+
+export function connection(socket: Socket) {
+  socket.on('joinExternalImageCapture', async (msg, callback) => {
+    if (!ensureProps(msg, ['variant_id', 'variant_token', 'answer_name'])) {
+      return callback(null);
+    }
+
+    if (!checkToken(msg.variant_token, msg.variant_id)) {
+      return callback(null);
+    }
+
+    socket.join(`variant-${msg.variant_id}-answer-${msg.answer_name}`);
+
+    const externalImageCaptureData: StatusMessage = {
+      variant_id: msg.variant_id,
+      answer_name: msg.answer_name,
+    };
+
+    callback(externalImageCaptureData);
```

Comment:
Nit, avoiding a local variable where we can.

```suggestion
    callback({
      variant_id: msg.variant_id,
      answer_name: msg.answer_name,
    } satisfies StatusMessage);
```